modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/util/IgniteMath.java

```diff
@@ -270,6 +272,15 @@ public static byte divideExact(byte x, byte y) {
         return (byte) (x / y);
     }
 
+    /** Decimal division. */
+    public static BigDecimal divideDecimal(BigDecimal x, BigDecimal y, int scale) {
+        if (x == null || y == null) {
+            return null;
+        } else {
+            return x.divide(y, new MathContext(scale, RoundingMode.HALF_EVEN));
```

Comment:
why `RoundingMode.HALF_EVEN`?

Reply:
We use HALF_EVEN (see a roundingMode in IgniteSqlFunctions) for decimal division.

Reply:
well, it's `RoundingMode.HALF_UP` (https://github.com/apache/ignite-3/blob/9592dd85d50b79e2baf03c16e8f2d0029c5a84d1/modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/IgniteSqlFunctions.java#L49)